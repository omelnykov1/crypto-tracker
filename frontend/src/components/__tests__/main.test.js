import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { configure } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import MainPage from '../main/main_page';

configure({ adapter: new Adapter() });
import { shallow } from 'enzyme';

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});


describe('<MainPage />', () => {
   const Wrapper = shallow(<MainPage />);
  it('should be defined', () => {
    expect(MainPage).toBeDefined()
  })
  it("should render", () => {
    expect(toJson(Wrapper)).toMatchSnapshot()
    expect(Wrapper.hasClass("main")).toEqual(true);
  });
  it("shows welcome message", () => {
    expect(Wrapper.find('h1').text()).toEqual('Welcome to Crypto Tracker')
  })

  it("has all coints button", () => {
    expect(Wrapper.find('button').text()).toEqual('All Coins')
  })

  it('has searchbar', () => {
    expect(Wrapper.find('<SearchBar />'))
  })
})

