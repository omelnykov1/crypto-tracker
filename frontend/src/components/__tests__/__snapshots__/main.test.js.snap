// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> should render 1`] = `
<div
  className="main"
>
  <div
    className="main-left"
  >
    <div
      className="welcome-message"
    >
      <h1>
        Welcome to Crypto Tracker
      </h1>
      <p
        className="welcome-main"
      >
        Platform which provides fast and reliable data about your favorite blockchain projects. The simplified overview of everything makes it easy to quickly navigate to the coins you have an interest in.
      </p>
      <div
        className="all-tickers"
      >
        <button
          className="tickers-btn"
          onClick={[Function]}
        >
          All Coins
        </button>
      </div>
    </div>
  </div>
  <div
    className="main-right"
  >
    <img
      alt=""
      src="/images/not-main.png"
    />
    <SearchBar />
  </div>
</div>
`;
